src/commands/release-asset/delete.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright 2025 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Command, Flags } from '@oclif/core';
+import { getBalenaSdk, stripIndent } from '../../utils/lazy';
+import { commitOrIdArg } from '../release';
+
+export default class ReleaseAssetDeleteCmd extends Command {
+	public static description = stripIndent`
+		Delete a release asset.
+
+		Delete a release asset with the specified key. This action cannot be undone.
+	`;
+
+	public static examples = [
+		'$ balena release-asset delete 1234567 --key config.json',
+		'$ balena release-asset delete a777f7345fe3d655c1c981aa642e5555 --key app.tar.gz',
+		'$ balena release-asset delete 1234567 --key old-asset --yes',
+	];
+
+	public static args = {
+		commitOrId: commitOrIdArg({
+			description: 'the commit or ID of the release',
+			required: true,
+		}),
+	};
+
+	public static flags = {
+		key: Flags.string({
+			description: 'the key of the release asset to delete',
+			required: true,
```

Comment:
Same question as in a previous comment about how we determine whether a required value should be an arg or flag

Reply:
Again I don't think the CLI has a strong guidance over it. Both (key and release) are mandatory. I think having two args is not ideal (as they are positional and can't be passed using preferred key/value). As release is mandatory in all 4 commands and key in only 3, I preferred the first as an argument (so that all commands feel similar like `balena release-asset list 123` `balena release-asset upload 123 ...`  ... and the key only on the commands that require it (e.g. list does not)